SimPEG/VRM/ProblemVRM.py

```diff
@@ -0,0 +1,609 @@
+from SimPEG import Problem, mkvc, Maps, Props, Solver as simpegSolver
+from SimPEG.VRM.SurveyVRM import SurveyVRM
+# from SimPEG.VRM.FieldsVRM import Fields_LinearFWD
+import numpy as np
+import scipy.sparse as sp
+import discretize
+
+############################################
+# BASE VRM PROBLEM CLASS
+############################################
+
+
+class BaseProblemVRM(Problem.BaseProblem):
+    """
+Base class for VRM problem.
+
+REQUIRED ARGUMENTS:
+
+    mesh: 3D tensor or OcTree mesh
+
+KWARGS:
+
+    refFact: Maximum refinement factor for sensitivities (default = 3)
+    refRadius: Distances from source in which cell sensitivities are refined.
+               Must be an array or list with elements equal to the refFact.
+               (default based on minimum cell size)
+    topoMap: A Maps object which operates from mesh cells to cells which are
+             computed in the forward model (default is all mesh cells)
+
+    """
+
+    # SET CLASS ATTRIBUTES
+    _refFact = None
```

Comment:
These can be set using the properties library fairly cleanly. 

```python
refFact = properties.Integer(
    "refinement factor",
    default = 3,
    min = 0
)
```

This will allow you to remove lines 43, 49, 64. To then do the additional checks in 66 - 70, you can use

```python
@properties.validator('refFact')
def check_vs_radius(self, change):
    val = change['value']
    assert val == len(self.refRadius), "Refinement factor no longer matches length of refinement radii array. Please ensure refinement factor is equal or less to number of elements in refinement radii"
    if val > 4:
         print("Refinement factor larger than 4 may result in computations which exceed memory limits")
```

cc @rowanc1

Reply:
I like things how I have them right now. But if I can get a detailed explanation as to why this is better, and it does the exact same thing, we could end up changing it.